Refactor pages related to garage-sign (#1705)
advancedtelematic/aktualizr: docs/ota-client-guide/modules/ROOT/pages/metadata-expiry.adoc

====
endif::[]

If you xref:rotating-signing-keys.adoc[take the keys for signing metadata offline], you need to refresh the metadata expiry date before it is reached. Otherwise, you won't be able to push updates. You can also define your own expiry dates when you take your keys offline.

The default expiry dates are as follows:

* For `targets.json`, the expiry date is **31 days** from the last update.

> * For `root.json`, the expiry date is **365 days** from the last update.

Author: Changed

> * For `root.json`, the expiry period is **365 days** from the last time a software update was created.

Reviewer: This is more wrong than the previous version. 

* An expiry isn't a period, it's a specific date/time. When you use garage-sign to create or modify metadata, it automatically updates the expiry date to a date that is 31/365 days in the future from the moment the command is run. Before the expiry date, the metadata is valid; after the expiry date it's expired.
* Creating a software update doesn't do anything at all to Uptane metadata.

Author: Changed. Hope I got it right.

== Define your own expiry dates

Depending on how you use OTA Connect, set the expiry dates in one of the following ways:

> * If you build software images, update your *build configuration*.

Reviewer: ```suggestion
* If you build software images using Yocto and our `meta-updater` layer, update your *build configuration*.
```

> * If you build software images, update your *build configuration*.
> * If you use the standalone command-line tools such as xref:install-garage-sign-deploy.adoc[`garage-deploy`], add extra *command-line arguments*.

Reviewer: ```suggestion
* If you build software images using bitbake with our `meta-updater` Yocto layer, update your *build configuration*.
* If you use xref:garage-sign-reference.adoc[`garage-sign`] to manage your metadata, add extra *command-line arguments*.
```

garage-deploy isn't actually relevant here.

Author: Deleted.

--
include::partial$config-descriptions.adoc[tags=buildconfig-hint]

> .Parameters of build configuration for metadata expiry

Reviewer: The previous wording is more standard.
```suggestion
.Build configuration parameters for metadata expiry
```

Author: Changed.
